MotionMark/tests/resources/main.js

```diff
@@ -681,104 +681,102 @@ class RampController extends Controller {
     }
 }
 
-Stage = Utilities.createClass(
-    function()
+class Stage {
+    constructor()
     {
-    }, {
+    }
 
-    initialize: function(benchmark)
+    initialize(benchmark)
     {
         this._benchmark = benchmark;
         this._element = document.getElementById("stage");
         this._element.setAttribute("width", document.body.offsetWidth);
         this._element.setAttribute("height", document.body.offsetHeight);
         this._size = GeometryHelpers.elementClientSize(this._element).subtract(Insets.elementPadding(this._element).size);
-    },
+    }
 
     get element()
     {
         return this._element;
-    },
+    }
 
     get size()
     {
         return this._size;
-    },
+    }
 
-    complexity: function()
+    complexity()
     {
         return 0;
-    },
+    }
 
-    tune: function()
+    tune()
     {
         throw "Not implemented";
-    },
+    }
 
-    animate: function()
+    animate()
     {
         throw "Not implemented";
-    },
+    }
 
-    clear: function()
+    clear()
     {
         return this.tune(-this.tune(0));
     }
-});
 
-Utilities.extendObject(Stage, {
-    random: function(min, max)
+    static random(min, max)
     {
         return (Pseudo.random() * (max - min)) + min;
-    },
+    }
 
-    randomBool: function()
+    static randomBool()
```

Comment:
These random functions can be moved to a separate class named Random or as extension to Math. They should not be tied to Stage.